crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1557,7 +1557,7 @@ pub async fn add_card_to_locker(
                 card_reference,
             )
             .await
-            .map_err(|error| {
+            .inspect_err(|_| {
```

Comment:
instead discarding the error, log there error also. Do the same for the below inspect_err also.

Reply:
These errors won't be discarded exactly, they will be returned implicitly by `inspect_err()`. The errors will only be ignored for the purpose of `inspect_err()`, because the error information is not needed here for raising metrics.